designs/0006-labels.md

```diff
@@ -0,0 +1,318 @@
+
+| Label                                               | AKS (documented) | AKS (observed or code) | Karpenter (schedulable) | Karpenter (written to node) | Notes |
+| --------------------------------------------------- | ---------------- | ---------------------- | ----------------------- | --------------------------- | ----- |
+| karpenter.sh/capacity-type                          |                  |                        | ✅                      | ✅                          |       |
+| karpenter.azure.com/sku-cpu                         |                  |                        | ✅                      | ✅                          |       |
+| karpenter.azure.com/sku-memory                      |                  |                        | ✅                      | ✅                          |       |
+| karpenter.azure.com/sku-gpu-name                    |                  |                        | ✅                      | ✅                          |       |
+| karpenter.azure.com/sku-gpu-manufacturer            |                  |                        | ✅                      | ✅                          |       |
+| karpenter.azure.com/sku-gpu-count                   |                  |                        | ✅                      | ✅                          |       |
+| karpenter.azure.com/sku-name                        |                  |                        | ✅                      | ✅                          |       |
+| karpenter.azure.com/sku-family                      |                  |                        | ✅                      | ✅                          |       |
+| karpenter.azure.com/sku-version                     |                  |                        | ✅                      | ✅                          |       |
+| karpenter.azure.com/sku-storage-ephemeralos-maxsize |                  |                        | ✅                      | ✅                          |       |
+| karpenter.azure.com/sku-storage-premium-capable     |                  |                        | ✅                      | ✅                          |       |
+| karpenter.azure.com/sku-networking-accelerated      |                  |                        | ✅                      | ✅                          |       |
+| karpenter.azure.com/sku-hyperv-generation           |                  |                        | ✅                      | ✅                          |       |
+
+#### Conclusion
+We will add these labels to WellKnownLabels + requirements:
+* kubernetes.azure.com/scalesetpriority
+* kubernetes.azure.com/node-image-version
+* kubernetes.azure.com/fips_enabled
+* kubernetes.azure.com/os-sku
+* kubernetes.azure.com/mode
+
+We will consider adding these labels to WellKnownLabels + requirements in the future:
+
+* kubernetes.azure.com/ebpf-dataplane - we may need to add this to requirements in the future to support network-dataplane updates. In the migration from azure -> cilium,
```

Comment:
Calling this out here we do not currently allow network transitions so this isn't an issue today, but if we wanted to allow network transitions it ends up leading into this conversation and I _think_ at least one solution (IMO probably the best solution) ends up putting `kubernetes.azure.com/ebpf-dataplane` in well-known labels, which would remove the need for the user to correct set this value on all of their NodePools based on the network topology of the cluster (we effectively set it for them).